ophyd/v2/core.py

```diff
@@ -145,6 +143,9 @@ class Device(HasName):
     #: The parent Device if it exists
     parent: Optional[Device] = None
 
+    def __init__(self, name: str = "") -> None:
+        self.set_name(name)
```

Comment:
why is this no longer needed? Does HasName have a constructor similar to this anyway?

Reply:
The constructor doesn't appear in the HasName protocol. I'm adding this here rather than in StandardReadable because the `epicsdemo.SampleStage` isn't actually `Readable`, so adding this allows it to inherit from `Device` rather than `StandardReadable`